src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final MachineProducer<Bulldozer> bulldozerMachineProducer = new BulldozerProducer();
+    private final MachineProducer<Excavator> excavatorMachineProducer = new ExcavatorProducer();
+    private final MachineProducer<Truck> truckMachineProducer = new TruckProducer();
+
+    @Override
+    public List<? extends Machine> getAll(Class type) {
+
+        if (type == Bulldozer.class) {
+            return bulldozerMachineProducer.get();
+        }
+        if (type == Excavator.class) {
+            return excavatorMachineProducer.get();
+        }
+        if (type == Truck.class) {
+            return truckMachineProducer.get();
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> list) {
+        for (Machine machine : list) {
+            machine.doWork();
+        }
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
```

Comment:
```suggestion
    public void fill(List<? extends Machine> machines, Machine value) {
```

Reply:
![image](https://user-images.githubusercontent.com/98239700/217612958-bca7ea71-fb13-44b0-9754-438a7f5c94cf.png)
This does not work and does not pass tests

Reply:
`fiil()` should be super. Everything is okay.